### ISSUE TITLE
TrapExit changes in sbt 0.13.1-RC3 make `run` SIGSEGV the JVM on Mac OS X

### ISSUE THREAD

**Issue by SethTisue** [sbt/sbt#997]
my code does use AWT classes, but it is intended to be run with `java.awt.headless=true`. I'm not forking, just using regular `run` or `run-main`

the crash happens with both Java 7 and Java 8. it started with RC3; RC2 was fine.

the details of the crash differ a little depending on whether `-Djava.awt.headless=true` is included or not. with:

```
Stack: [0x00007fff5c2aa000,0x00007fff5c32a000],  sp=0x00007fff5c327160,  free space=500k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [liblwawt.dylib+0xd1df]  setUpAppKitThreadName+0x55
C  [liblwawt.dylib+0xd8f9]  -[AWTStarter starter:]+0xb9
C  [Foundation+0x640de]  __NSThreadPerformPerform+0xe5
C  [CoreFoundation+0x7f8f1]  __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__+0x11
C  [CoreFoundation+0x71062]  __CFRunLoopDoSources0+0xf2
C  [CoreFoundation+0x707ef]  __CFRunLoopRun+0x33f
C  [CoreFoundation+0x70275]  CFRunLoopRunSpecific+0x135
C  [java+0x73b0]  CreateExecutionEnvironment+0x367
C  [java+0x1b5c]  JLI_Launch+0x7a0
C  [java+0x770d]  main+0x65
C  [java+0x13b4]  start+0x34
```

and without `headless=true`:

```
Stack: [0x00007fff5cd01000,0x00007fff5cd81000],  sp=0x00007fff5cd7df10,  free space=499k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [JavaNativeFoundation+0x4b59]  JNFNSToJavaString+0x6d
C  [libosxapp.dylib+0x37df]  +[PropertiesUtilities javaSystemPropertyForKey:withEnv:]+0x1b
C  [libosxapp.dylib+0x1b01]  -[NSApplicationAWT registerWithProcessManager]+0x106
C  [libosxapp.dylib+0x1493]  -[NSApplicationAWT init]+0x8c
C  [AppKit+0x36d1]  +[NSApplication sharedApplication]+0x91
C  [liblwawt.dylib+0xd94a]  -[AWTStarter starter:]+0x10a
C  [Foundation+0x640de]  __NSThreadPerformPerform+0xe5
C  [CoreFoundation+0x7f8f1]  __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__+0x11
C  [CoreFoundation+0x71062]  __CFRunLoopDoSources0+0xf2
C  [CoreFoundation+0x707ef]  __CFRunLoopRun+0x33f
C  [CoreFoundation+0x70275]  CFRunLoopRunSpecific+0x135
C  [java+0x73b0]  CreateExecutionEnvironment+0x367
C  [java+0x1b5c]  JLI_Launch+0x7a0
C  [java+0x770d]  main+0x65
C  [java+0x13b4]  start+0x34
```

these particular stack traces are with

```
# JRE version: Java(TM) SE Runtime Environment (7.0_45-b18) (build 1.7.0_45-b18)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (24.45-b08 mixed mode bsd-amd64 compressed oops)
```

**Comment by SethTisue** [comment]
this program is sufficient to trigger the crash:

```
object O {
  def main(argv: Array[String]) {
    new java.awt.Color(0,0,0)
  }
}
```

**Comment by SethTisue** [comment]
note that for now this can be worked around with `trapExit in Global := false`

**Comment by harrah** [comment]
It will be included in the soon-to-be-published 0.13.1-RC5, which is RC5 because I published RC4 without this fix.
